server/controllers/userController.js

```diff
@@ -1,35 +1,56 @@
 import bcrypt from 'bcrypt';
+import pool from '../database/dbConnection';
+import { createUser } from '../database/sqlQueries';
 import user from '../database/user';
 import auth from '../helpers/auth';
 import validator from '../helpers/validator';
 
 class UserController {
   static create(req, res) {
-    const { email } = req.body;
-    const users = user.findAll();
-    if (users.find(entry => entry.email === email)) {
-      return res.status(409).json({
-        status: 409,
-        error: 'Email already exist'
-      });
-    }
-    const userObj = req.body;
-    userObj.password = bcrypt.hashSync(req.body.password, 10);
-    userObj.isAdmin = false;
-    userObj.type = 'client';
-    const refData = user.create(userObj);
-    if (refData) {
-      const { password, ...data } = refData;
-      data.token = auth.createToken({ id: data.id, type: data.type, isAdmin: data.isAdmin });
-      return res.status(201).json({
-        status: 201,
-        data,
+    const {
+      email, firstName, lastName,
+      stateOfResidence, phoneNumber, title, password, dateOfBirth, sex
+    } = req.body;
+
+    const params = [
+      email,
+      firstName,
+      lastName,
+      bcrypt.hashSync(password, 10),
+      stateOfResidence,
+      phoneNumber,
+      title,
+      dateOfBirth,
+      sex
+    ];
+
+    pool.query(createUser, params)
+      .then((queryData) => {
+        const createdUser = queryData.rows[0];
+        const { id, type, isAdmin } = createdUser;
+        const token = auth.createToken({ id, type, isAdmin });
+        const { password: pass, ...data } = createdUser;
+        data.token = token;
+        return res.status(201).json({
+          status: 201,
+          data: [data]
+        });
+      })
+      .catch((err) => {
+        console.log(err);
+        if (err.code === '23505') {
```

Comment:
what does this error code `23505` mean?

Reply:
It is used to detect when there is a violation of constraint, the table has a constraint of unique on the email column, I use the code to tell if a user tries to submit an email that is already on the database